File: exts/jphp-gui-game-ext/src/main/java/org/develnext/jphp/ext/game/classes/UXGameBackground.java
package org.develnext.jphp.ext.game.classes;

import javafx.scene.Node;
import javafx.scene.image.Image;
import javafx.scene.layout.AnchorPane;
import org.develnext.jphp.ext.game.GameExtension;
import org.develnext.jphp.ext.game.support.GameBackground;
import org.develnext.jphp.ext.game.support.GamePane;
import org.develnext.jphp.ext.game.support.Vec2d;
import org.develnext.jphp.ext.javafx.classes.UXCanvas;
import org.develnext.jphp.ext.javafx.classes.layout.UXPanel;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Namespace;
import php.runtime.annotation.Reflection.Nullable;
import php.runtime.annotation.Reflection.Property;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Namespace(GameExtension.NS)
public class UXGameBackground extends UXCanvas<GameBackground> {
    interface WrappedInterface {
        @Property @Nullable Image image();
        @Property Vec2d velocity();
        @Property("viewPosition") Vec2d viewPos();

        @Property boolean autoSize();
    }

    public UXGameBackground(Environment env, GameBackground wrappedObject) {
        super(env, wrappedObject);
    }

    public UXGameBackground(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Override
    public void __construct() {
        __wrappedObject = new GameBackground();
    }

    public GameBackground getWrappedObject() {
        return (GameBackground) super.getWrappedObject();
    }
}

File: exts/jphp-gui-game-ext/src/main/resources/JPHP-INF/sdk/php/game/UXGameBackground.php
<?php
namespace php\game;

use php\gui\UXCanvas;
use php\gui\UXImage;

/**
 * Class UXGameBackground
 * @package php\game
 * @packages game, javafx
 */
class UXGameBackground extends UXCanvas
{
    /**
     * --RU--
     * Изображение.
     * @var UXImage
     */
    public $image;

    /**
     * --RU--
     * Линейная скорость [x, y].
     * @var array
     */
    public $velocity = [0, 0];

    /**
     * --RU--
     * Позиция вида [x, y].
     * @var array
     */
    public $viewPosition = [0, 0];

    /**
     * --RU--
     * Авторазмер.
     * @var bool
     */
    public $autoSize = false;

    /**
     * --RU--
     * Отразить изображение по X.
     * @var bool
     */
    public $flipX = false;

    /**
     * --RU--
     * Отразить изображение по Y.
     * @var bool
     */
    public $flipY = false;
}